siriuspy/siriuspy/devices/scraper.py

```diff
@@ -0,0 +1,311 @@
+        # check if device exists
+        if devname not in _ScraperDev.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        if properties is None:
+            properties = _ScraperDev._properties
+
+        super().__init__(devname, properties)
+
+    @property
+    def is_force_cmd_complete(self):
+        """."""
+        return self['ForceComplete-Mon']
+
+    @property
+    def is_coordinate_convertion_ok(self):
+        """."""
+        return self['CoordConvErr-Mon']
+
+    @property
+    def is_backlash_compensation_enabled(self):
+        """."""
+        return self['EnblBacklashComp-Sts']
+
+    def cmd_enable_backlash_compensation(self):
+        """."""
+        self['EnblBacklashComp-Sel'] = 1
+
+    def cmd_disable_backlash_compensation(self):
+        """."""
+        self['EnblBacklashComp-Sel'] = 0
+
+    def cmd_go_max_aperture(self):
+        """Go to maximum slit aperture."""
+        self['Home-Cmd'] = 1
+
+
+class ScraperH(_ScraperDev):
+    """."""
+
+    _properties = (
+        'OuterSlitPos-SP', 'OuterSlitPos-RB',
+        'InnerSlitPos-SP', 'InnerSlitPos-RB',
+        'InnerDoneMov-Mon', 'OuterDoneMov-Mon',
+        'OuterSlitInnerLim-SP', 'OuterSlitInnerLim-RB',
+        'OuterSlitOuterLim-SP', 'OuterSlitOuterLim-RB',
+        'OuterSlitBacklashDist-SP', 'OuterSlitBacklashDist-RB',
+        'OuterMotionCtrl-Cte', 'ForceOuterSlitPos-Cmd',
+        'InnerSlitInnerLim-SP', 'InnerSlitInnerLim-RB',
+        'InnerSlitOuterLim-SP', 'InnerSlitOuterLim-RB',
+        'InnerSlitBacklashDist-SP', 'InnerSlitBacklashDist-RB',
+        'InnerMotionCtrl-Cte', 'ForceInnerSlitPos-Cmd',
+        )
+
+    def __init__(self):
+        """."""
+        # call base class constructor
+        _all_properties = _ScraperDev._properties + ScraperH._properties
+        super().__init__(
+            devname=_ScraperDev.DEVICES.H, properties=_all_properties)
+
+    @property
+    def left_slit(self):
+        """Left slit position [mm]."""
+        return self['OuterSlitPos-RB']
+
+    @property
+    def right_slit(self):
+        """Right slit position [mm]."""
+        return self['InnerSlitPos-RB']
+
+    @property
+    def is_left_slit_moving(self):
+        """."""
+        return not self['OuterDoneMov-Mon']
+
+    @property
+    def is_right_slit_moving(self):
+        """."""
+        return not self['InnerDoneMov-Mon']
+
+    @property
+    def left_slit_limits(self):
+        """Limits for left slit.
+
+        Left Slit (x>0)             Right Slit (x<0)
+        [Outer, Inner] | (x=0) | [Inner, Outer]
+        """
+        return [self['OuterSlitOuterLim-RB'], self['OuterSlitInnerLim-RB']]
+
+    @left_slit_limits.setter
+    def left_slit_limits(self, limits):
+        """Set left slit limits.
+
+        Left Slit (x>0)             Right Slit (x<0)
+        [Outer, Inner] | (x=0) | [Inner, Outer]
+        """
+        self['OuterSlitInnerLim-SP'] = limits[0]
+        self['OuterSlitOuterLim-SP'] = limits[1]
```

Comment:
there is an inconsistency here: 
getter returns outer and inner
setter receives inner and outer

Reply:
(assuming the bug pointed out above has been fixed) why are the limits sorted in descending order?

Reply:
Yes that was a bug, thanks Ana. 

The definition was confusing indeed, as Fernando also pointed in the comment above. Now the limits are always `[inner, outer]`, where `inner` refers to the minimum distance from the slit to the beam and `outer` refers to the maximum distance. If we ignore the sign of values (inner and bottom slits the values are typically negative), now the limits are sorted in ascending order. Please take a look in the updated code.